### DISCUSSION TITLE
Unexpected behavior when string literals contain }}

### DISCUSSION THREAD

**Discussion by jolleekin** [harttle/liquidjs#290] [Q&A] [answered]
### Input
```liquid
{{ '{{' }}
{{ '}}' }}
```

### [Actual output](https://liquidjs.com/playground.html#e3sgJ3t7JyB9fQp7eyAnfX0nIH19)
```
{{
' }}
```

### Expected output
```
{{
}}
```

The issue is at [tokenizer.ts, line 133](https://github.com/harttle/liquidjs/blob/master/src/parser/tokenizer.ts#L133)

**Comment by harttle** [comment]
It's by design. To output raw string literals, use [raw](https://liquidjs.com/tags/raw.html):

```liquid
{% raw %}{{{% endraw %}
{% raw %}}}{% endraw %}
```

**Comment by jolleekin** [comment, marked as answer]
I consider this as a limitation/design flaw of the tokenizer. This Dart package [liquid_engine](https://pub.dev/packages/liquid_engine) can handle this case without any problem.

In general, the biggest problem with Liquid is that it doesn't have an official spec to address syntax, parsing, and rendering. This undesirably allows different implementations to have different behaviors.

**Reply by harttle** [reply]
> the biggest problem with Liquid is that it doesn't have an official spec

You're right. I'll mark it as a bug and try to fix it. I believe no one wants `{{` in string literals be parsed as a delimiter, let's try to make the syntax more intuitive!

**Reply by harttle** [reply]
Done! checkout liquidjs@9.20.1

**Reply by jolleekin** [reply]
Thanks. That was quick! :)
